test/staged.jl

```diff
@@ -156,7 +156,7 @@ end
 @generated function _g_f_with_inner(x)
     :(y->y)
 end
-@test (_g_f_with_inner(1))(8) == 8
+@test_throws ErrorException _g_f_with_inner(1)
 
```

Comment:
Yes, I wouldn't have made this an error if that code wasn't already broken, so that this was just adding a clearer error than "segfault", corruption of precompiled files, or other errors in which method gets called.

I look forward to having anonymous functions again, so that they are uniquely defined by content rather than by name.